lib/publiccloud/instance.pm

```diff
@@ -476,7 +481,9 @@ sub wait_for_ssh {
 
         # Finally make sure that SSH works
         while (($duration = time() - $start_time) < $args{timeout}) {
-            $exit_code = $self->ssh_script_run(cmd => "true", username => $args{username}, timeout => $args{timeout} - $duration);
+            # After the instance is resumed from hibernation the SSH can freeze
+            my $ssh_opts = $self->ssh_opts() . ' -o ControlPath=none -o ConnectTimeout=10';
+            $exit_code = script_run("ssh -v $ssh_opts -l $args{username} $self->{public_ip} true", timeout => $args{timeout} - $duration);
             last if isok($exit_code);
```

Comment:
I "warmly" suggest to move this line on top as first command into this loop, because if already isok, no check is needed at all and also because this way it takes into account the previous `exit_code` value elaborated before.

Reply:
The previous `ssh` command runs with `-o StrictHostKeyChecking=no` and this runs with `-o StrictHostKeyChecking=yes`.  That's why I'd like to do this check so we're sure the key verification works.

The order was also the same before. I didn't change it.

Maybe we can have a call about it?

Reply:
This loop was already in **ERROR** before, I was hoping you fixed it.

Reply:
Thank you much for our clarifying call; after all consideration discussed, please feel free to proceed with your updates.